mtags/src/main/scala/scala/meta/internal/pc/Completions.scala

```diff
@@ -1050,14 +1067,25 @@ trait Completions { this: MetalsGlobal =>
         val lzy =
           if (sym.isLazy) "lazy "
           else ""
+
+        val newKeyword = defn
+          .map {
+            case _: DefDef => "def"
+            case _ => "val"
+          }
+          .getOrElse("def")
+
         val edit = new l.TextEdit(
           range,
-          s"${overrideKeyword}${lzy}${keyword} $label = $${0:???}"
+          s"${overrideKeyword}${lzy}${newKeyword} $label = $${0:???}"
         )
+
+        // if we had no val or def then filter will be empty
+        val fullFilter = (if (defn.isDefined) filter else prefix) + sym.name.decoded
```

Comment:
The filter text controls how the editor filters out completion items client-side as the user types more characters. 

![2019-04-06 11 37 00](https://user-images.githubusercontent.com/1408093/55667710-56549800-5860-11e9-8294-d01b11fac713.gif)

- on `override` completion, filter text should be `"override def $name"`
- on `def` completion, filter text should be `"def name"`
- on `val` completion, filter text should be `"val name"`